src/JustSaying/Messaging/MessageProcessingStrategies/IMessageProcessingStrategy.cs

```diff
@@ -38,5 +38,10 @@ public interface IMessageProcessingStrategy
         /// will be available immediately when <see cref="StartWorkerAsync"/> is subsequently called.
         /// </remarks>
         Task<int> WaitForAvailableWorkerAsync();
+
+        /// <summary>
+        /// Notifies of a successful message received, waits for any debounce/circuit break logic.
+        /// </summary>
+        Task ReportMessageReceived(bool success);
```

Comment:
I'm finding the method name `ReportMessageReceived` confusing, it's less about reporting anything, it's more about introducing an optional wait. Perhaps `WaitForListenerThrottling(bool success)` or something? (that's not a great name either I know 😆)

Reply:
Agreed that a `WaitFor...` would be better. I think the Listener might be confused with SqsNotificationListener, so maybe just `WaitForThrottlingAsync` ?

Reply:
Yes, that's a good fit